### ISSUE TITLE
Unexpected token trying to build tensorflow r0.10.0rc0

### ISSUE THREAD

**Issue by mirosval** [bytedeco/javacpp-presets#264]
I'm stuck trying to compile the javacpp-presets for the new tensorflow. I get the following output:

```
[INFO]
[INFO] --- maven-resources-plugin:2.7:resources (javacpp.parser) @ tensorflow ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.3:compile (javacpp.parser) @ tensorflow ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to javacpp-presets/tensorflow/target/classes
[INFO]
[INFO] --- javacpp:1.2.2:build (javacpp.parser) @ tensorflow ---
[INFO] Detected platform "macosx-x86_64"
[INFO] Building for platform "macosx-x86_64"
[INFO] Targeting javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/default/integral_types.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/framework/numeric_types.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/init_main.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/types.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/mutex.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/macros.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/util/port.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/bazel-genfiles/tensorflow/core/lib/core/error_codes.pb.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/logging.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/lib/core/status.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/protobuf.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/file_system.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/tensorflow/core/platform/env.h
[INFO] Parsing javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/bazel-genfiles/tensorflow/core/protobuf/config.pb.h
[ERROR] Failed to execute JavaCPP Builder: javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/bazel-genfiles/tensorflow/core/protobuf/config.pb.h:201: Unexpected token '::'
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] JavaCPP Presets .................................... SUCCESS [  0.857 s]
[INFO] JavaCPP Presets for TensorFlow ..................... FAILURE [12:09 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 12:10 min
[INFO] Finished at: 2016-08-10T11:36:45+02:00
[INFO] Final Memory: 29M/400M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.bytedeco:javacpp:1.2.2:build (javacpp.parser) on project tensorflow: Failed to execute JavaCPP Builder: javacpp-presets/tensorflow/cppbuild/macosx-x86_64/tensorflow-0.10.0rc0/bazel-genfiles/tensorflow/core/protobuf/config.pb.h:201: Unexpected token '::' -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :tensorflow
```

The offending piece of code:
```
 private:

  ::google::protobuf::internal::InternalMetadataWithArena _internal_metadata_;
  friend class ::google::protobuf::Arena; // <- line 201
  typedef void InternalArenaConstructable_;
```

Tensorflow compiles w/o problems using bazel, so I'm not sure exactly what I can do to fix this.

Tensorflow 0.10.0 is right around the corner, so I think it will help you as well to get it to work as soon as possible in order to keep up to date.

**Comment by saudet** [comment]
Yes, there's something weird that changed with protobuf v3.0.0. You could try to debug this if you can, it would help :)

**Comment by mirosval** [comment]
I could take a look, but I have no idea where to start. Since this is happening after Bezel compilation has finished successfully, I'm assuming this is a bug in some javacpp parser? Do you have any pointers for me where to start?

**Comment by saudet** [comment]
Yes, the parser. It's pretty crude, but it's just a hand written recursive descent parser, so it's easy to interpret with debuggers...

**Comment by saudet** [comment]
It's all in this file: https://github.com/bytedeco/javacpp/blob/master/src/main/java/org/bytedeco/javacpp/tools/Parser.java

**Comment by mirosval** [comment]
The parser was easy to find, but I can't figure out how to even run the javacpp.jar from the commandline on the javacpp-presets tensorflow project, let alone in the IDE. I couldn't find any good documentation that would explain what goes where in those flags, what needs to be on classpath and so on.

I tried to infer it from the pom.xml but that is very confusing as well. 

I understand that I need to pass the tensorflow headers, the tensorflow.so file and the javacpp-presets tensorflow files (compiled .classes or jar? or .java?) but I haven't been able to figure out how to pass all that to javacpp.jar without it ending up with some ClassDefNotFound exception.

**Comment by saudet** [comment]
Try to use `mvnDebug`, that's going to let you attach remotely any Java debugger.

**Comment by mirosval** [comment]
I ended up running with

`mvn compile -Djavacpp.cppbuild.skip=true` and `mvnDebug compile -Djavacpp.cppbuild.skip=true`

Pushed a PR here: #266

**Comment by sonalgupta** [comment]
@mirosval I forked your fork of the javacpp repo and tried to run `mvn compile -Djavacpp.cppbuild.skip=true --projects .,tensorflow` but it still gives the same error `[ERROR] Failed to execute JavaCPP Builder: /private/var/tmp/_bazel_sonal/f9d07472005cef91813a467bf5eb7506/execroot/tensorflow-0.10.0/bazel-out/local-opt/genfiles/tensorflow/core/protobuf/config.pb.h:201: Unexpected token '::'` 

Is this expected?

**Comment by mirosval** [comment]
Following these steps:

1. Clone and compile master of javacpp with `mvn install`
2. Clone javacpp-presets
3. in javacpp-presets run `./cppbuild.sh -platform macosx-x86_64 install tensorflow`
4. in javacpp-presets run `mvn install --projects .,tensorflow`

I don't get this error.

**Comment by nchivukula** [comment]
@mirosval 

I am trying to build javacpp-presets for tensorflow r0.10.0rc0. Followed the instructions above after I cloned from your fork. I am getting following error, did you happen to encounter this issue:


[INFO] Reactor Summary:
[INFO] 
[INFO] JavaCPP Presets .................................... SUCCESS [  0.971 s]
[INFO] JavaCPP Presets for TensorFlow ..................... FAILURE [ 20.874 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 23.067 s
[INFO] Finished at: 2016-09-28T06:28:23-07:00
[INFO] Final Memory: 34M/409M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (javacpp.parser) on project tensorflow: Compilation failure: Compilation failure:
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1419,33] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1421,33] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1441,29] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1443,29] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1462,31] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1464,31] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1478,40] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1480,35] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.Env
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1051,40] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1053,35] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1056,37] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1058,37] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1061,39] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1063,39] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1066,40] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1068,35] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.FileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1132,40] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1134,35] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1137,29] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1139,29] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1142,31] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1144,31] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1148,10] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] /home/samsung/dev/serving/javacpp-presets/tensorflow/src/main/java/org/bytedeco/javacpp/tensorflow.java:[1151,10] cannot find symbol
[ERROR] symbol:   class UniquePtr
[ERROR] location: class org.bytedeco.javacpp.tensorflow.NullFileSystem
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (javacpp.parser) on project tensorflow: Compilation failure
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:911)
	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:129)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)

**Comment by mirosval** [comment]
Make sure you're using the latest javacpp presets, this has been fixed there but not yet released AFAIK.

**Comment by nchivukula** [comment]
Thanks!! I was able to build with the snapshot version of javacpp.

**Comment by saudet** [comment]
Fix included in JavaCPP 1.2.4. Thanks for reporting and fixing this!
